engine/src/org/pentaho/di/trans/Trans.java

```diff
@@ -308,10 +308,10 @@
   private volatile boolean preparing;
 
   /** Whether the transformation is initializing. */
-  private boolean initializing;
+  private volatile boolean initializing;
```

Comment:
@mbatchelor I suggest using Enums for execution state and execution phase: `enum ExecutionState { RUNNING, STOPPED, PAUSED}` and `enum ExecutionPhase {PREPARING, INITIALIZING, EXECUTING, STOPPED, FINISHED}` and use them instead of AtomicInteger fields because it  will help to avoid  redundant code to check if the number we set to this AtomicInteger values is valid. So what do you think about it?